### PR TITLE
MULE-12140 Modifing api to support expected output and split

### PR DESCRIPTION
**Pull request by machaval**

### CODE REVIEW

File: core-tests/src/test/java/org/mule/runtime/core/el/DataWeaveExpressionLanguageAdaptorTestCase.java [hunk 1]
```diff
@@ -103,9 +93,29 @@ public void attributesBinding() throws Exception {
   }
 
   @Test
+  public void splitByJson() throws Exception {
+    Event jsonMessage = eventBuilder().message(Message.builder().payload("[1,2,3]").mediaType(MediaType.APPLICATION_JSON).build()).build();
```
Comment (afelisatti) [inline core-tests/src/test/java/org/mule/runtime/core/el/DataWeaveExpressionLanguageAdaptorTestCase.java]: Use static imports for constants (APPLICATION_JSON, BOOLEAN).

File: core/src/main/java/org/mule/runtime/core/el/mvel/MVELSplitDataIterator.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  public static Iterator<TypedValue<?>> createFrom(Object result, int bachSize) {
+    Iterator<Object> iter;
+    if (result instanceof Object[]) {
+      iter = Arrays.asList((Object[]) result).iterator();
+    } else if (result instanceof Iterable<?>) {
+      iter = ((Iterable<Object>) result).iterator();
+    } else if (result instanceof Iterator<?>) {
+      iter = (Iterator<Object>) result;
+    } else if (result instanceof Map<?, ?>) {
+      iter = ((Map) result).values().iterator();
```
Comment (afelisatti) [inline core/src/main/java/org/mule/runtime/core/el/mvel/MVELSplitDataIterator.java]: Why iterate the values and not the entries?

File: core/src/main/java/org/mule/runtime/core/el/mvel/MVELSplitDataIterator.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    } else if (result == null) {
+      iter = Collections.emptyIterator();
```
Comment (afelisatti) [inline core/src/main/java/org/mule/runtime/core/el/mvel/MVELSplitDataIterator.java]: Static imports.

File: core-tests/src/test/java/org/mule/runtime/core/el/DataWeaveExpressionLanguageAdaptorTestCase.java [hunk 4]
```diff
@@ -202,36 +212,12 @@ public void variablesCannotOverrideEventBindings() throws MuleException {
...
     assertThat(expressionLanguage.evaluate(ERROR, event, BindingContext.builder().build()).getValue(), is(nullValue()));
     assertThat(expressionLanguage.evaluate(VARIABLES, event, BindingContext.builder().build()).getValue(),
-               is(instanceOf(Map.class)));
+            is(instanceOf(Map.class)));
     assertThat(expressionLanguage.evaluate("flow.name", event, mockFlowConstruct, BindingContext.builder().build()).getValue(),
-               is(flowName));
-  }
-
-  @Test
-  public void accessRegistryBean() throws MuleException {
```
Comment (afelisatti) [inline core-tests/src/test/java/org/mule/runtime/core/el/DataWeaveExpressionLanguageAdaptorTestCase.java]: Why delete this tests?

File: core/src/main/java/org/mule/runtime/core/el/mvel/MVELSplitDataIterator.java (hunk 2 above)
Comment (machaval) [inline core/src/main/java/org/mule/runtime/core/el/mvel/MVELSplitDataIterator.java]: Agree but this is how it used to be in the expression spliter. For me I would prefer to iterate over the entires

File: core-tests/src/test/java/org/mule/runtime/core/el/DataWeaveExpressionLanguageAdaptorTestCase.java (hunk 4 above)
Comment (machaval) [inline core-tests/src/test/java/org/mule/runtime/core/el/DataWeaveExpressionLanguageAdaptorTestCase.java]: Ups this looks like something went wrong

File: core-tests/src/test/java/org/mule/runtime/core/el/DataWeaveExpressionLanguageAdaptorTestCase.java [hunk 5]
```diff
@@ -20,9 +21,8 @@
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.spy;
 import static org.mockito.Mockito.when;
-import static org.mule.runtime.api.metadata.DataType.OBJECT;
-import static org.mule.runtime.api.metadata.DataType.STRING;
-import static org.mule.runtime.api.metadata.DataType.fromType;
+import static org.mule.runtime.api.metadata.DataType.*;
```
Comment (afelisatti) [inline core-tests/src/test/java/org/mule/runtime/core/el/DataWeaveExpressionLanguageAdaptorTestCase.java]: Remove the star import.

**Pull request merged**